lms/djangoapps/instructor/views/data_access.py

```diff
@@ -77,10 +77,25 @@ def get_group_query_students(course_id, group_id):
     Asynchronously makes the subqueries for a group and then aggregates them once the subqueries are finished.
     """
     _group, queries, _relation = get_saved_queries(course_id, group_id)
-    #wait for the queries to finish before proceeding
+    # wait for the queries to finish before proceeding
     make_subqueries.apply_async(args=(course_id, group_id, queries)).get()
-    queried_students = retrieve_grouped_query(course_id, group_id)
-    return queried_students
+    # forcing evaluation here because we will be deleting stuff afterward
+    subqueries_qs = GroupedTempQueryForSubquery.objects.filter(grouped_id=group_id).distinct().values_list('query',
+                                                                                                           flat=True)
+    if subqueries_qs:
+        queried_students_qs = retrieve_grouped_query(course_id, group_id)
+        if queried_students_qs:
+            # calling len to force evaluation because we will delete intermediate queries
+            len(queried_students_qs)
```

Comment:
This is not needed because it gets evaluated on the if statement above.
https://docs.djangoproject.com/en/1.7/ref/models/querysets/#when-querysets-are-evaluated